### PR TITLE
Fix: #1292 — Allow overwriting default OpenAPI response descriptions

### LINKED ISSUE #1292: Enhancement: Add a way to override default OpenAPI response descriptions
**What's the feature you'd like to ask for.**
Currently there is no way to replace the default OpenAPI description for HTTP 400 errors. For example, you currently can't do this:

```python
@starlite.post(
    "/path",
    responses={
        400: starlite.ResponseSpec(
            model=MyErrorModel, description="A more detailed error description"
        )
    },
)
async def login() -> starlite.Response:
    ...
```

If you do, you will get this error on startup:

```console
starlite.exceptions.http_exceptions.ImproperlyConfiguredException: 500: Additional response for status code 400 already exists in success or error responses
```

This makes it impossible to add more detailed error descriptions or properly document response shapes if you want your endpoint to return a different error shape.

**Additional context**
Removing this bit of code would allow the default HTTP 400 description to be replaced:
https://github.com/seladb/starlite/blob/51ae7eebda3e6834a6520471bfdb9cad2ab790c5/starlite/openapi/responses.py#L217-L220

### PR DESCRIPTION
**Pull request by Nadock**
# PR Checklist

- [x] Have you followed the guidelines in `CONTRIBUTING.rst`?
- [x] Have you got 100% test coverage on new code?
- [ ] Have you updated the prose documentation?
- [ ] Have you updated the reference documentation?

Closes #1292

### CODE REVIEW

**Review by Goldziher — COMMENTED** [review]
Hi, first off thanks for contributing - second, please add tests.

**Comment by Nadock** [comment]
> Hi, first off thanks for contributing - second, please add tests.

I've updated the preexisting test that covered this case previously to be more rigorous. If there is other scenarios you think are worth covering in tests let me know.

**Review by JacobCoffee — APPROVED** [review]

File: tests/openapi/test_responses.py [hunk 1]
```diff
@@ -249,15 +248,15 @@ def test_additional_responses_overlap_with_other_responses() -> None:
...
-    with pytest.raises(
-        ImproperlyConfiguredException,
-        match="Additional response for status code 200 already exists in success or error responses",
-    ):
-        create_responses(handler, raises_validation_error=True, generate_examples=False, plugins=[])
+    responses = create_responses(handler, raises_validation_error=True, generate_examples=False, plugins=[])
+
+    assert responses is not None
+    assert responses["200"] is not None
+    assert responses["200"].description == "Overwritten response"
```
Comment (Goldziher) [inline tests/openapi/test_responses.py:259]: please add one more test- 

we need to test raises and then customizing the response

Comment (Nadock) [reply tests/openapi/test_responses.py:259]: be4dca8

**Review by Goldziher — APPROVED** [review]

**Comment by Goldziher** [comment]
@all-contributors add @Nadock code

**Pull request merged**